Eastern-Research-Group/mywaterway
# Feature/add usgs streamgages

## Related Issues:
* TBD (@cschwinderg I'm having Login.gov issues so can't access EPA's Jira account right now, but will update this with the issue number once I'm able to login)

## Main Changes:
* Initial implementation of adding USGS Streamgages to Community Overview tab. The biggest change is the Community Overview tab now displays sub tabs for "Waterbodies" (previously existing, but now in their own tab), "Monitoring & Sensors" (new USGS streamgages, and existing monitoring stations that were previously only displayed on the Monitoring tab), and "Permitted Dischargers" (newly displayed in a listview). There's still a few outstanding questions EPA needs to answer re: the specifics of what's displayed for the streamgages, and how they're colored, but this lays most of the groundwork for what will be displayed.

## Steps To Test:
1. Navigate to http://localhost:3000/community/020700100301/overview
1. Switch tabs to "Monitoring & Sensors"
1. Toggle "Daily Stream Flow Conditions" and "Monitoring Stations" and ensure the listview below (along with its title) changes to reflect toggle state
1. Hover a UGSG Streamgage in the listview and ensure corresponding point on map shows hovered state.
1. Expand a UGSG Streamgage in the listview, click "View on Map" button, and ensure map zooms to point and map popup displays with same info as what's shown in the listview
1. Do the same two above steps from the map (hover and select USGS Streamgage point on map and ensure corresponding listview item is shown as hovered/selected, with map popup showing when it's selected)
1. Expand the map legend and ensure legend is displayed for USGS Stremgages when they're displayed on map
1. Change location on the map to see Sample Locations update.

## TODO:
- [x] Properly link toggling of "Sample Locations" switch to rendering of USGS streamgages and monitoring stations on map (probably move the `updateVisibleLayers` implementation of those layers into an effect that depends on the toggle state – e.g. `usgsStreamgagesDisplayed` and `monitoringStationsDisplayed`)
- [x] Merge and normalize the properties of USGS streamgages and monitoring stations so they can be properly displayed in on list that can be sortable via a common set of "Sort By" fields.
- [x] Determine if there should be some other visual difference between the two types of sample locations (streamgages vs monitoring stations) in the listview (e.g. maybe a "tag" that's displayed?). Right now you can tell at a glance something is a streamgage if it doesn't have a value for "Monitoring Measurements" as that's not returned in the USGS web service, but it might be a good idea to differentiate between the two visually in the listview
- [x] Create popups for the USGS streamgages w/ the same contents displayed in a listview (I saw how this was done for other components, but before spreading code across too many files, I figured it'd be best to get the display of the listview squared away first – just easier to maintain code in one place initially)
- [x] Update the first number in the "# of #" in the `AccordionList` title to reflect the number of sample locations displayed via the switches above
- [x] Move the USGS web service URL into the `services.json` file
- [x] Update the `usgsStreamgagesLayer` to be a FeatureLayer, so we set its render to be dynamic via the `visualVariables` property, and display each streamgage's gage height colorized, similar to PurpleAir monitors.
- [x] ~~After using a FeatureLayer instead of GraphicsLayer, we'll need to have two renderers if we want to display the numeric value along with a binned color (e.g. a number inside a colorized circle)~~ **UPDATE:** Ended up using a FeatureLayer label instead, which is a much better solution
- [ ] Determine what sort of parameter code mapping is needed so we can display tailored text for each parameter in a streamgage's listview, instead of the description returned from the USGS web service (**NOTE:** We'll discuss this with EPA in our next meeting, so no need to wait on this before merging this in)
